integrationTests/testObserverConnector.go

```diff
@@ -0,0 +1,143 @@
+package integrationTests
+
+import (
+	"errors"
+	"fmt"
+	"math/rand"
+	"time"
+
+	wsData "github.com/multiversx/mx-chain-communication-go/websocket/data"
+	wsFactory "github.com/multiversx/mx-chain-communication-go/websocket/factory"
+	"github.com/multiversx/mx-chain-core-go/data/outport"
+	"github.com/multiversx/mx-chain-core-go/marshal"
+	logger "github.com/multiversx/mx-chain-logger-go"
+	"github.com/multiversx/mx-chain-notifier-go/api/shared"
+	"github.com/multiversx/mx-chain-notifier-go/common"
+	"github.com/multiversx/mx-chain-notifier-go/config"
+	"github.com/multiversx/mx-chain-notifier-go/factory"
+	"github.com/multiversx/mx-chain-notifier-go/process"
+)
+
+// CreateObserverConnector will create observer connector component
+func CreateObserverConnector(facade shared.FacadeHandler, connType string, apiType string) (ObserverConnector, error) {
+	marshaller := &marshal.JsonMarshalizer{}
+	switch connType {
+	case common.HTTPConnectorType:
+		return NewTestWebServer(facade, apiType), nil
+	case common.WSObsConnectorType:
+		return NewTestWSServer(connType, facade, marshaller)
+	default:
+		return nil, errors.New("invalid observer connector type")
+	}
+}
+
+// NewTestWSServer will create a new test ws server
+func NewTestWSServer(connType string, facade process.EventsFacadeHandler, marshaller marshal.Marshalizer) (ObserverConnector, error) {
+	port := getRandomPort()
+	conf := config.WebSocketConfig{
+		URL:                "localhost:" + fmt.Sprintf("%d", port),
+		WithAcknowledge:    true,
+		Mode:               "server",
+		RetryDurationInSec: 5,
+		BlockingAckOnError: false,
+	}
+
+	_, err := factory.CreateWSObserverConnector(connType, conf, marshaller, facade)
+	if err != nil {
+		return nil, err
+	}
+
+	// wait for ws server to start
+	time.Sleep(2 * time.Second)
+
+	wsClient, err := NewWSObsClient(marshaller, conf.URL)
+	if err != nil {
+		return nil, err
+	}
+
+	// wait for ws client to start
+	time.Sleep(2 * time.Second)
+
+	return wsClient, nil
+}
+
+func getRandomPort() int {
+	min := 22111
+	max := 22222
+	return rand.Intn(max-min) + min
+}
+
+type wsObsConnector struct {
```

Comment:
Why the need for this struct if it's the same as `wsObsClient`?

Reply:
that's actually not being used, removed it